eahub/base/views.py

```diff
@@ -94,28 +94,77 @@ def groups(request):
     )
 
 
-def get_groups_data():
-    rows = Group.objects.all()
+def get_groups_data(user, by_distance=False):
+    if hasattr(user, "profile") and by_distance:
+        if hasattr(user.profile, "lat"):
```

Comment:
This check doesn't work; all profiles have a `lat` attribute. You want `if user.profile.lat is not None`.

Also, what happens if the user has a profile but it doesn't have a location? I think this would crash in that case.

Also also, I think you should check for both `lat` and `lon`; normally they should go together, but this isn't strictly enforced. (I mean, it is on users, but not on staff.)